src/ServiceControl.Audit.Persistence.RavenDB/CustomChecks/CheckDirtyMemory.cs

```diff
@@ -0,0 +1,103 @@
+namespace ServiceControl.Audit.Persistence.RavenDB.CustomChecks;
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NServiceBus.CustomChecks;
+using NServiceBus.Logging;
+
+class CheckDirtyMemory(MemoryInformationRetriever memoryInformationRetriever) : CustomCheck("ServiceControl.Audit database", "Dirty memory trends", TimeSpan.FromMinutes(5))
+{
+    readonly List<int> lastDirtyMemoryReads = [];
+    public override async Task<CheckResult> PerformCheck(CancellationToken cancellationToken = default)
+    {
+        var (isHighDirty, dirtyMemoryKb) = await memoryInformationRetriever.GetMemoryInformation(cancellationToken);
+
+        if (isHighDirty)
+        {
+            var message = $"There is a high level of dirty memory ({dirtyMemoryKb}kb). Check the ServiceControl " +
```

Comment:
Since this is user visible, it feels like it should say something like "There is a high level of RavenDB dirty memory"